Mathlib/RingTheory/Flat/Basic.lean

```diff
@@ -229,6 +246,122 @@ instance of_projective [h : Projective R M] : Flat R M := by
   cases h with
     | _ e he => exact of_retract R _ _ _ _ he
 
+open BigOperators in
+lemma rTensor_preserves_injectiveness_of_injective_characterModule
+    (h : Module.Injective R <| CharacterModule M) :
+    Flat.rTensor_preserves_injectiveness R M := by
+  intros A B _ _ _ _ L hL
+  rw [← LinearMap.ker_eq_bot, eq_bot_iff]
+  rintro z (hz : _ = 0)
+  show z = 0
+  by_contra rid
+  obtain ⟨g, hg⟩ := CharacterModule.exists_character_apply_ne_zero_of_ne_zero (a := z) rid
+
+  let f : A →ₗ[R] (CharacterModule M) :=
```

Comment:
Can you separate this definition from the proof?

Reply:
I shortened it to one line